src/charm.py

```diff
@@ -206,12 +221,14 @@ def __init__(self, *args):
     def self_scraping_job(self):
         """The self-monitoring scrape job."""
         external_url = urlparse(self._external_url)
-        return [
-            {
-                "metrics_path": f"{external_url.path}/metrics",
-                "static_configs": [{"targets": [f"{external_url.hostname}:{external_url.port}"]}],
-            }
-        ]
+        job = {
+            "metrics_path": f"{external_url.path}/metrics",
+            "static_configs": [{"targets": [f"{external_url.hostname}:{external_url.port}"]}],
+        }
+        if self.server_cert.cert:
+            job["scheme"] = "https"
+
+        return [job]
```

Comment:
`external_url.hostname` is provided by `Traefik`, for instance:

![imagen](https://github.com/canonical/alertmanager-k8s-operator/assets/939888/7479c8e6-ab8a-495b-a2ca-ba470fe11fec)


On the other hand `self.server_cert.cert` is provided by `CertManager`.

If `Traefik` doesn't support `HTTPS` in the `URLs` it provides, we can end up generating an invalid URL

Reply:
Yep. And I think we said self-scrape is going to be FQDN?
Anyway, let's leave this for #165? I added your comment [there](https://github.com/canonical/alertmanager-k8s-operator/issues/165#issuecomment-1603307584).